### PR TITLE
Sonarlint updates to SubscriptionModelMapper class.

### PR DESCRIPTION
**Pull request by CanadianPhilipBarkow**
### Description

Null pointer check added to `SubscriptionModelMapper`.

Relevant test updates made.

### Related Azure Boards Work Items

[AB#4173](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4173)
[AB#4195](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4195)

### Checklist

- [X] I have tested the changes locally
- [X] I have added/updated tests that prove my fix is effective or that my feature works
- [X] I have checked that my code follows the project's coding style by running `npm run format:check`
- [X] I have checked that my code contains no linting errors by running `npm run lint`
- [X] I have checked that my code contains no type errors by running `npm run typecheck`
- [X] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [X] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java [hunk 1]
```diff
@@ -25,6 +25,9 @@ public abstract class SubscriptionModelMapper extends AbstractModelMapper {
 
 	@Nullable
 	public CollectionModel<SubscriptionModel> toModel(String userId, @Nullable Iterable<Subscription> subscriptions) {
+		if (subscriptions == null || subscriptions.spliterator() == null) {
```
Comment (gregory-j-baker) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java:28]: `subscriptions.spliterator()` will never be null here

```suggestion
		if (subscriptions == null) { return CollectionModel.empty(); }
```

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import ca.gov.dtsstn.cdcp.api.service.domain.Subscription;
+import ca.gov.dtsstn.cdcp.api.service.domain.ImmutableSubscription;
+
+@ExtendWith({ MockitoExtension.class })
+class SubscriptionModelMapperTest {
+
+	SubscriptionModelMapper subscriptionModelMapper = Mappers.getMapper(SubscriptionModelMapper.class);
+
+	@Test
+	final void testToModelStringIterableOfSubscriptionNullSubscription() {
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java:24]: You should add a `@DisplayName()` annotation here

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		final Collection<Subscription> subscriptions = null;
+		assertThat(subscriptionModelMapper.toModel("d827416b-f808-4035-9ccc-7572f3297015", subscriptions)).isEmpty();
+	}
+	@Test
+	final void testToModelStringIterableOfSubscriptionNonNullSubscription() {
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java:29]: - You should add a `@DisplayName()` annotation here
- You should add a space between methods for enhanced readability

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		final Collection<Subscription> subscriptions = Stream.of(ImmutableSubscription.builder().id("00000000-0000-0000-0000-000000000000").build(), ImmutableSubscription.builder().id("22222222-2222-2222-2222-222222222222").build()).collect(Collectors.toCollection(HashSet::new));
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java:30]:
```suggestion
		final var subscriptions = List.of(
			ImmutableSubscription.builder()
				.id("00000000-0000-0000-0000-000000000000")
				.build(),
			ImmutableSubscription.builder()
				.id("22222222-2222-2222-2222-222222222222")
				.build());
```

(for readability)

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+		assertThat(subscriptionModelMapper.toModel("d827416b-f808-4035-9ccc-7572f3297015", subscriptions)).isNotEmpty();
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java:31]: You should run some assertions on the values returned. The mapper could return a list of `[null, null, null, null]` and this test would still pass.

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java (hunk 4 above)
Comment (CanadianPhilipBarkow) [reply backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java]: Implemented in commit [9fbe7e7](https://github.com/DTS-STN/canadian-dental-care-plan/pull/1865/commits/9fbe7e70e363ba40945580d5841081219b4bf781).

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java (hunk 3 above)
Comment (CanadianPhilipBarkow) [reply backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java]: Good idea, implemented in commit [9fbe7e7](https://github.com/DTS-STN/canadian-dental-care-plan/pull/1865/commits/9fbe7e70e363ba40945580d5841081219b4bf781).

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java (hunk 2 above)
Comment (CanadianPhilipBarkow) [reply backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java]: Implemented in commit commit [9fbe7e7](https://github.com/DTS-STN/canadian-dental-care-plan/pull/1865/commits/9fbe7e70e363ba40945580d5841081219b4bf781).

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java (hunk 1 above)
Comment (CanadianPhilipBarkow) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java]: Superfluous check removed in commit [9fbe7e7](https://github.com/DTS-STN/canadian-dental-care-plan/pull/1865/commits/9fbe7e70e363ba40945580d5841081219b4bf781).

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java [hunk 6]
```diff
@@ -21,14 +19,27 @@ class SubscriptionModelMapperTest {
...
+					.id("00000000-0000-0000-0000-000000000000")
+					.build(),
+				ImmutableSubscription.builder()
+					.id("22222222-2222-2222-2222-222222222222")
+					.build());
+		final var subscriptionModels = subscriptionModelMapper.toModel("d827416b-f808-4035-9ccc-7572f3297015", subscriptions);
+		assertThat(subscriptionModels)
+			.hasSize(2)
+			.anyMatch(model -> model.getId() == "00000000-0000-0000-0000-000000000000")
+			.anyMatch(model -> model.getId() == "22222222-2222-2222-2222-222222222222");
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapperTest.java:42]: Strings have to be compared with `.equals()`

(🤣 doesn't sonarlint warn you about this? 🤣)

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**